### ISSUE TITLE
nuxt generate crawler fails after nuxt 3.12

### ISSUE THREAD

**Issue by DallasHoff** [nuxt/content#2664]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.10.0`
- Nuxt Version:     `3.12.1`
- CLI Version:      `3.12.0`
- Nitro Version:    `2.9.6`
- Package Manager:  `npm@10.8.1`
- Builder:          `-`
- User Config:      `css`, `modules`
- Runtime Modules:  `@nuxt/content@2.12.1`
- Build Modules:    `-`

### Reproduction

[Repo](https://github.com/DallasHoff/nuxt-fouc-example) (disregard the repo name; it's for a different issue)

1. Run `npx nuxi@latest module add content`
2. Run `nuxt generate`
3. Observe that `.output/public/index.html` exists since Nuxt 3.11 is installed
4. Run `npx nuxi@latest upgrade --force` to upgrade Nuxt to 3.12
5. Run `nuxt generate` again
6. Observe that `.output/public/index.html` does not exist

### Describe the bug

After upgrading to Nuxt 3.12, `nuxt generate` outputs no `index.html` or HTML files for any other pages defined in the `pages` directory if the Nuxt Content module is installed. 

There is no error displayed in the output of `nuxt generate`. It simply reports that the crawler only found 3 routes: `200.html`, `404.html`, and `/api/_content/cache.1718155374568.json`. If `@nuxt/content` is removed from `nuxt.config.ts`, `nuxt generate` will find and render the rest of the pages.

**Comment by sshwy** [comment]
facing the same issue

**Comment by farnabaz** [comment]
According to this comment, it is a regression in Nuxt 3.12. https://github.com/nuxt/nuxt/issues/27490#issuecomment-2162890462

Until it fixes in the upstream you can add this route rule into your config.

```ts
export default defineNuxtConfig({
  routeRules: {
    '/': {
      prerender: true
    }
  },
})
```

**Comment by isaackearl** [comment]
> According to this comment, it is a regression in Nuxt 3.12. [nuxt/nuxt#27490 (comment)](https://github.com/nuxt/nuxt/issues/27490#issuecomment-2162890462)
> 
> Until it fixes in the upstream you can add this route rule into your config.
> 
> ```ts
> export default defineNuxtConfig({
>   routeRules: {
>     '/': {
>       prerender: true
>     }
>   },
> })
> ```

This doesn't work for me unfortunately.  If I downgrade to Nuxt 3.11 it renders 19 routes.  If I upgrade to Nuxt 3.12 I get 3 but they don't render correctly.  Using the routeRules didn't seem to make a difference for me, am I supposed to put every route that I use in there? (am I doing it wrong?)

**Comment by DallasHoff** [comment]
The new 3.12.2 patch of nuxt fixed this.

**Comment by isaackearl** [comment]
> The new 3.12.2 patch of nuxt fixed this.

This issue is not fixed for me in 3.12.2.  

I cannot get all my routes to prerender as they normally do when I generate.  I'm running a client side only (ssr: false), project where we always build to production staticly.  

Without making any modifications in 3.11 when I run npm run generate, it works great and prerenders 19 routes.  the static generation works great etc.. then I upgrade to 3.12.2, and it renders 4 routes, but they only seem to be the default 404 and things like that... none of my actual pages.  One note is that our software doesn't have a page at the root "/", as we are using nuxt to replace old legacy pages in an older application.  This hasn't been a problem in any previous version of Nuxt.

![image](https://github.com/nuxt/content/assets/9354830/99739a09-e087-4681-bbbf-c2fd627b1b0c)

On 3.11 I have 19 initial routes which includes all my actual pages I've created and it doesn't have the / route (which is correct as that route doesn't exist in my project)
